File: Tutorial4-8LayerLibrary/src/main/java/com/test/tutorial4_8layerlibrary/viewmodel/BaseViewModel.java
package com.test.tutorial4_8layerlibrary.viewmodel;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.test.tutorial4_8layerlibrary.model.User;


public class BaseViewModel extends ViewModel {

    private MutableLiveData<String> stringMutableLiveData = new MutableLiveData<>();

    private MutableLiveData<User> userMutableLiveData = new MutableLiveData<>();

    public BaseViewModel() {

    }
}

File: Tutorial4-8LayerLibrary/src/main/java/com/test/tutorial4_8layerlibrary/viewmodel/BaseViewModel.kt
package com.test.tutorial4_8layerlibrary.viewmodel

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.test.tutorial4_8layerlibrary.model.User

class BaseViewModel : ViewModel() {
    private val stringMutableLiveData = MutableLiveData<String>()
    private val userMutableLiveData = MutableLiveData<User>()
}